aesd-char-driver/main.c

```diff
@@ -220,18 +222,152 @@ ssize_t aesd_write(struct file *filp, const char __user *ubuf, size_t count, lof
         /* Loop in case multiple '\n' exist in the now-updated incomplete_cmd */
     }
 
+   
+
   out_unlock_free:
+    if (retval > 0)                 // only advance when we actually wrote bytes
+        *f_pos += retval;           // keep positional I/O consistent with llseek
     mutex_unlock(&dev->lock);
     kfree(kbuf);
     return retval;
 }
+
+static loff_t aesd_buffer_size_bytes(const struct aesd_circular_buffer *buf)
```

Comment:
Would be good to note that caller must hold lock on this function.  Also might be good to move this into the circular buffer source file.